Automattic/wp-calypso
# Domains: Add new "Disconnect" button for domain management pages

Adds a "Disconnect" button to the domain management page - the button should convert a regular domain to a domain-only site:

![image](https://github.com/Automattic/wp-calypso/assets/18705930/47e6c22f-3cfd-44ea-9079-161023a7d5b8)

## Testing Instructions
- Click on the "disconnect" button;
- Click on "disconnect" button on the confirmation dialog;
- Ensure that the domain will be converted to a domain-only site and that the navigation looks correct;
- Additionally, ensure that the "disconnect" button/section/card isn't displayed for domain-only sites.

## Pre-merge Checklist
